### PR TITLE
feat: add seedlot navigator in seedlot dashboard

### LINKED ISSUE #1626: Feature: easier navigation to seedlot registration from my seedlots page
**As a** *Orchard manager/technician* **I want** *a feature that allows me to navigate to a seedlot* **So That** *can quickly navigate to the correct registration form*

**Acceptance Criteria**
- [ ] Given that I know my seedlot number, When I enter it, Then I am able to easily navigate to it.

**Additional Context**
- Originally, the table in the seedlot dashboard was supposed to help users navigate to the right seedlot, however the paginated results only allow for searching to occur on one screen.

**Definition of Done**
- [ ] Ready to Demo in Sprint Review
- [ ] Does what I have made have appropriate test coverage?
- [ ] Documentation and/or scientific documentation exists and can be found
- [ ] Peer Reviewed by 2 people on the team
- [ ] QA from a UX 
- [ ] Manual testing of all PRs in Dev and Prod
- [ ] Merged

### PR DESCRIPTION
**Pull request by xiaopeng0202**
# Description
Closes #1626 

#### New
- added go to seedlot button in seedlot dashboard and my seedlot page

#### Changed
- refactored and created seedlot navigator component

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-43-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1643-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1643-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-43-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1643-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1643-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Review by mgaseta — CHANGES_REQUESTED** [review]
I think this task was to add the seedlot navigator to "My Seedlots" page, not to the "Seedlot Dashboard", can you confirm @SLDonnelly and @kevinginley ?

**Comment by kevinginley** [comment]
> I think this task was to add the seedlot navigator to "My Seedlots" page, not to the "Seedlot Dashboard", can you confirm @SLDonnelly and @kevinginley ?

Yes that's right @mgaseta

**Comment by xiaopeng0202** [comment]
> I think this task was to add the seedlot navigator to "My Seedlots" page, not to the "Seedlot Dashboard", can you confirm @SLDonnelly and @kevinginley ?

We will add the component in both my seedlot and seedlot dashboard page

**Comment by kevinginley** [comment]
@xiaopeng0202 @mgaseta 

On my screen there are some finicky alignment issues (fyi I'm using Chrome)

image

image

Ideally the right edge of the "Go to seedlot" button would align with the right edge of the table, whereas here it is spilling outside of the table, which added a scrollbar to the bottom of my screen.

At the smallest breakpoint:
image
- could you please add some padding between the "Go to seedlot" button and the field above it
- and align the left edge of the "Go to seedlot" button with the left edge of the "Register a new seedlot"

As I'm playing around with the responsiveness, is there a way to consistently make the right edge of the "Register a new seedlot" button align with the right edge of the "Go to seedlot" button
image

Thank you!

File: frontend/src/utils/NumberUtils.ts [hunk 1]
```diff
@@ -7,3 +7,5 @@ export const isFloatWithinRange = (value: string, min: string, max: string): boo
   new BigNumber(value).isGreaterThanOrEqualTo(min)
   && new BigNumber(value).isLessThanOrEqualTo(max)
 );
+
+export const isNumeric = (value: string) => /^-?\d+$/.test(value);
```
Comment (RMCampos) [inline frontend/src/utils/NumberUtils.ts:11]: suggestion (non-blocking): how about having a simpler change and more readable? I'm thinking of something like this: `export const isNumeric = (value: string) => value.replace(/[0-9]/g, '').length === 0;`

**Review by RMCampos — CHANGES_REQUESTED** [review]
Great job! Just a small suggestion for the regex and some styling issues :) 💯

**Comment by RMCampos** [comment]
Styling issues

- My Seedlots

Can we improve this component on "My Seedlots" page? I can see it kind of sticking to the left menu, looking quite weird.
image

- Fix title and subtitle according to page

I'm at the Seedlots page, and I still can see the review title and subtitle:

image

That's all =D great job!

**Comment by kevinginley** [comment]
The alignments are looking good, thank you!

Is there a way we could get consistent padding between the field and the "Go to seedlot button" between the different breakpoints?

Here we have quite a bit of space:
image

compared to:
image

But if it's going to be super tricky then I wouldn't worry about it.

**Comment by RMCampos** [comment]
@xiaopeng0202 nice. However I still can see "Review seedlot" at the Seedlots page.

**Review by RMCampos — CHANGES_REQUESTED** [review]
Need to fix the title.

**Review by RMCampos — APPROVED** [review]
All clear! Great job Xiao! 💯

**Pull request merged**